credoai/artifacts/data/comparison_data.py

```diff
@@ -113,15 +115,41 @@ def _validate_subjects_sensitive_features(self):
                         f"than one unique value. Only found one value: {unique_values[0]}"
                     )
 
+    def _preprocess_pairs(self):
+        """Preprocess the input `pairs` object"""
+        cols = ["source-subject-id", "target-subject-id"]
+        for col in cols:
+            self.pairs[col] = self.pairs[col].astype(str)
 
     def _preprocess_subjects_sensitive_features(self):
         """Preprocess the input `subjects_sensitive_features` object"""
-        sensitive_features_names = list(self.subjects_sensitive_features.columns)
-        sensitive_features_names.remove("subject-id")
-        for sf_name in sensitive_features_names:
-            self.subjects_sensitive_features[
-                sf_name
-            ] = self.subjects_sensitive_features[sf_name].astype(str)
+        if self.subjects_sensitive_features is not None:
+            cols = list(self.subjects_sensitive_features.columns)
+            for col in cols:
+                self.subjects_sensitive_features[
+                    col
+                ] = self.subjects_sensitive_features[col].astype(str)
```

Comment:
All of this can simply be:
`self.subjects_sensitive_features = self.subjects_sensitive_features.astype(str)`